src/utils/format.ts

```diff
@@ -132,6 +133,38 @@ function _decomposeBn(amount: BN, amountPrecision: number, decimals: number): {
   return { integerPart, decimalPart, decimalsPadded }
 }
 
+/**
+ * Transforms (possibly decimal) amount as string into a BN
+ * Takes into account original precision from amount + argument precision
+ *
+ *
+ * Internally we use BNs, but we all know BNs don't like decimals.
+ * To convert arbitrary strings, we need to make them integers.
+ * We do so by creating a BigNumber and moving the decimal separator to the right
+ * by multiplying it by 10^(amount precision).
+ * Which we then later add to regular precision to be 'compressed'.
+ * Finally, convert it to a string base 10,
+ * to avoid numbers with exponents which BN also doesn't like
+ *
+ * @param amountStr Amount with arbitrary precision
+ * @param amountPrecision precision on top of existing precision
+ */
+function _stringToBn(amountStr:string, amountPrecision:number):{ amount: BN, precision: number } {
+  const bigNumberAmount = new BigNumber(amountStr)
+
+  const originalPrecision = bigNumberAmount.precision()
+
+  const amount = new BN(bigNumberAmount
+    .multipliedBy(TEN_BIG_NUMBER
+      .pow(originalPrecision))
+    .integerValue()
+    .toString(10))
+
+  const precision = originalPrecision + (amountPrecision as number)
```

Comment:
Because I might want to keep the original precision.

I said in the description that I want to preserve the original precision and maybe add on top of it.

The use case is to deal with numbers that are decimals in the correct precision already, and I just need to properly format them.

With a further refactor of how we do the formatting this might be skipped entirely, but since right now we work with BNs, I need to add the original precision so it can be removed later and give me back the original value.

Example:
I want to nicely format the number `1234.56789`.
I expect to receive a string with decimal separator and with the expected amount of decimals, such as `1,234.5679`
There's no need to adjust the precision here. So I'll call `formatSmart('1234.56789', 0)`.

Internally (or externally like I'm currently doing) I need to first convert this string representation of a decimal number to BN.
Thus, I return the original precision together with the amount as BN, so I know how to format it back to it's original form.

That would result in `formatAmount(BN('123456789'), 5)`

Is that more clear?

Reply:
If it behaves like that, should we make it more clear in the param then: `precissionToAdd` so it's more clear.
seeing `_stringToBn(amount='123.45', precission=0)` I was initially expecting to return sth like `123` (so u know the precision is 0)

...or `_stringToBn(amount='123.4567', precission=5)` returning 12345670

so if u really add, maybe is a matter of changing the name to `amountPrecission` param?